pallets/nfts/src/lib.rs

```diff
@@ -233,25 +252,46 @@ impl<T: Config> NFTs for Pallet<T> {
     type AccountId = T::AccountId;
     type NFTDetails = T::NFTDetails;
     type NFTId = T::NFTId;
+    type NFTSeriesDetails = SeriesDetails;
 
     fn create(
         owner: &Self::AccountId,
         details: Self::NFTDetails,
+        series_details: Option<Self::NFTSeriesDetails>,
     ) -> result::Result<Self::NFTId, DispatchError> {
-        let id = Total::<T>::get();
-        Total::<T>::put(id.checked_add(&1.into()).ok_or(Error::<T>::NFTIdOverflow)?);
+        let nft_id = Total::<T>::get();
+        Total::<T>::put(
+            nft_id
+                .checked_add(&1.into())
+                .ok_or(Error::<T>::NFTIdOverflow)?,
+        );
+
         Data::<T>::insert(
-            id,
+            nft_id,
             NFTData {
                 owner: owner.clone(),
                 details,
                 sealed: false,
                 locked: false,
+                series_details: series_details.clone(),
             },
         );
 
-        Self::deposit_event(Event::Created(id, owner.clone()));
-        Ok(id)
+        if let Some(series_details) = series_details.as_ref() {
+            let mut array = sp_std::vec![];
+            let series_id = series_details.series_id;
+
+            if Series::<T>::contains_key(series_id) {
+                array = Series::<T>::get(series_id);
+            }
+
+            array.push(nft_id);
+            Series::<T>::insert(series_id, array);
```

Comment:
Based on this code and the whole function it seems like it would be possible for anyone to add NFTs to any series, we should probably permission this instead